static/js/timerender.ts

```diff
@@ -304,6 +305,28 @@ export function stringify_time(time: number | Date): string {
     return format(time, "h:mm a");
 }
 
+export function format_time_modern(time: number | Date, today = new Date()): String {
+    const hours = differenceInHours(today, time);
+    const days_old = differenceInCalendarDays(today, time);
+
+    const previous_year_leap = (today.getFullYear() - 1) % 4 === 0;
+    // If previous year is a leap year, same day will be 365 days ago and
+    // we want to display year in that case.
+    const a_year_has_passed = previous_year_leap ? days_old > 365 : days_old > 364;
```

Comment:
This logic is incorrect in multiple ways.

* January 15, 2019 + 365 days = January 15, 2020
* January 15, 2020 + 366 days = January 15, 2021
* March 15, 2019 + 366 days = March 15, 2020
* March 15, 2020 + 365 days = March 15, 2021